test/runtests.jl

```diff
@@ -2037,26 +2298,253 @@ end
+    y_bin = ifelse.(y .> median(y), 1.0, 0.0)
+    glm_fweights = glm(X, y_bin, Binomial(); wts=fweights(w))
+    glm_aweights = glm(X, y_bin, Binomial(); wts=aweights(Float64.(w)))
+
+    # Coefficients should match for frequency and analytic weights
+    @test coef(glm_fweights) ≈ coef(glm_aweights)
+end
+
+@testset "Weight conversion and deprecation" begin
+    X = [ones(10) randn(10)]
+    y = randn(10)
+
+    # Empty UnitWeights should silently expand to correct size (no warning)
+    # This is the internal default for formula-based fit methods
+    model = lm(X, y; wts=uweights(0))
+    @test GLM.weights(model) == uweights(10)
+
+    # Plain Vector weights should trigger deprecation warning and convert to FrequencyWeights
+    w = ones(10)
+    @test_logs (:warn, r"Passing weights as vector is deprecated") begin
+        model = lm(X, y; wts=w)
+        @test GLM.weights(model) isa FrequencyWeights
+    end
+end
+
+@testset "loglik_apweights_obs coverage" begin
+    # Test loglik_apweights_obs functions for various distributions
+    # These are used internally for analytic weights loglikelihood calculations
+
+    # Test Gamma with analytic weights
+    clotting = DataFrame(; u=log.([5, 10, 15, 20, 30, 40, 60, 80, 100]),
+                         lot1=[118.0, 58, 42, 35, 27, 25, 21, 19, 18],
+                         w=[1.5, 2.0, 1.1, 4.5, 2.4, 3.5, 5.6, 5.4, 6.7])
+
+    gm_gamma = glm(@formula(lot1 ~ u), clotting, Gamma(), InverseLink();
+                   wts=aweights(clotting.w), atol=1e-09, rtol=1e-09)
+    # Should compute without error and return valid loglikelihood
+    @test isfinite(loglikelihood(gm_gamma))
+    @test loglikelihood(gm_gamma) ≈ -43.359078787690514 rtol = 1e-06
+
+    # Test Geometric with analytic weights (NegativeBinomial(1) is equivalent)
+    quine = RDatasets.dataset("MASS", "quine")
+    quine.w = log.(3 .+ 3 .* quine.Days)
+
+    gm_geom = glm(@formula(Days ~ Eth + Sex), quine, Geometric(), LogLink();
+                  wts=aweights(quine.w), atol=1e-08, rtol=1e-08)
+    @test isfinite(loglikelihood(gm_geom))
+    # Value verified against R: glm(..., family=negative.binomial(theta=1, link="log"), weights=w)
+    @test loglikelihood(gm_geom) ≈ -2055.246 rtol = 1e-05
+
+    # Test InverseGaussian with analytic weights
+    gm_ig = glm(@formula(lot1 ~ u), clotting, InverseGaussian(), InverseSquareLink();
+                wts=aweights(clotting.w), atol=1e-09, rtol=1e-09)
+    @test isfinite(loglikelihood(gm_ig))
+    # Value verified against R: glm(..., family=inverse.gaussian(link="1/mu^2"), weights=w)
+    @test loglikelihood(gm_ig) ≈ -86.82547 rtol = 1e-05
+end
+
+@testset "Weight conversion function" begin
+    # Test convert_weights helper function
+    w = [1.0, 2.0, 3.0, 4.0, 5.0]
+    n = length(w)
+
+    # Vector should be converted to FrequencyWeights (with deprecation warning)
+    # Note: Base.depwarn is used internally, which behaves differently from @warn
```

Comment:
Deprecation is only tested above.
```suggestion
    # Vector should be converted to FrequencyWeights (with deprecation warning)
```